/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

package noisicaa.pb;

import "noisicaa/builtin_nodes/step_sequencer/model.proto";

message UpdateStepSequencer {
  required uint64 node_id = 1;
  optional uint32 set_num_steps = 2;
  optional bool set_time_synched = 3;
  optional uint32 add_channel = 4;
}

message UpdateStepSequencerChannel {
  required uint64 channel_id = 1;
  optional StepSequencerChannel.Type set_type = 2;
  optional float set_min_value = 3;
  optional float set_max_value = 4;
  optional bool set_log_scale = 5;
}

message DeleteStepSequencerChannel {
  required uint64 channel_id = 1;
}

message UpdateStepSequencerStep {
  required uint64 step_id = 1;
  optional bool set_enabled = 2;
  optional float set_value = 3;
}
